src/meta/src/barrier/command.rs

```diff
@@ -663,7 +670,19 @@ impl CommandContext {
             Command::CancelStreamingJob(table_fragments) => {
                 let node_actors = table_fragments.worker_actor_ids();
                 self.clean_up(node_actors).await?;
-                // Drop fragment info in meta store.
+
+                let table_id = table_fragments.table_id().table_id;
+                let mut table_ids = table_fragments.internal_table_ids();
+                table_ids.push(table_id);
+                if let Err(e) = self.hummock_manager.unregister_table_ids(&table_ids).await {
+                    tracing::warn!("Failed to unregister compaction group for {:#?}. They will be cleaned up on node restart. {:#?}", &table_ids, e);
+                }
+                self.catalog_manager
+                    .cancel_create_table_procedure_with_table_fragments(
```

Comment:
It can be called twice for normal creating jobs (not recovered ones). But it's acceptable to me since it's idempotent.